File: selfstudy/golang/students_sum.go
/*
task 211:
Students - Sum
You need to calculate the total sum of student test scores
*/

package main

import (
	"fmt"
	"reflect"
)

func TotalStudents(scores []int) int {
	sum := 0
	for _, v := range scores {
		sum += v
	}
	return sum
}

func main() {
	tests := []struct {
		input []int
		expected int
	}{
		{[]int{10, 20, 30}, TotalStudents([]int{10, 20, 30})},
		{[]int{50, 60, 70}, TotalStudents([]int{50, 60, 70})},
		{[]int{1, 2, 3}, TotalStudents([]int{1, 2, 3})},
		{[]int{100, 200}, TotalStudents([]int{100, 200})},
		{[]int{}, TotalStudents([]int{})},
	}
	
	for idx, t := range tests {
		res := TotalStudents(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// works i think

File: selfstudy/js/students_sum.js
/*
task 11:
Students - Sum
You need to calculate the total sum of student test scores
*/

function total_students(scores) {
  return scores.reduce((a, b) => a + b, 0);
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: total_students([10, 20, 30]) },
  { input: [50, 60, 70], expected: total_students([50, 60, 70]) },
  { input: [1, 2, 3], expected: total_students([1, 2, 3]) },
  { input: [100, 200], expected: total_students([100, 200]) },
  { input: [], expected: total_students([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('total_students', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(total_students(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, total_students(t.input) === t.expected || JSON.stringify(total_students(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// finally works
